cilium/CFP-39876-clustermesh-filtered-export.md

```diff
@@ -0,0 +1,226 @@
+
+## Motivation
+
+The existing Cilium ClusterMesh implementation distributes all CiliumIdentities and CiliumEndpoints across connected clusters, creating scaling constraints as resource volumes grow. This comprehensive propagation model becomes a limiting factor when managing extensive endpoint and identity collections. Performance profiling analysis demonstrates the severity of this scaling challenge: in a 200-cluster ClusterMesh deployment with the following configuration parameters, each Cilium agent consumes 15GiB of memory:
+
+- clusters=200
+- nodes/cluster=300
+- identities/cluster=500
+- endpoints/cluster=15000
+- services/cluster=0
+
+By restricting export to only resources under global namespaces, we can achieve substantial scalability improvements and operational efficiency gains. An important consideration for using this new global/local namespace feature is that all resources under the global namespaces are exported to the remote clusters. This implies that the memory and compute optimizations might be suboptimal if the cluster has most of the resources under the global namespaces. Without the proposed optimizations, each clustermesh-apiserver replica would export the entire local state to remote clusters.  
+
+## Goals
+
+- Enhance Cilium ClusterMesh scalability through selective endpoint and identity distribution
+- Full network policy support for pods inside global namespaces
+- Cross-cluster pod to pod connectivity inside global namespaces
+- Global service and MCS-API support for resources under global namespaces
+
+## Non-Goals
+
+- Network policies for non global(local) namespaces
+- Cross-cluster pod to pod connectivity for non global(local) namespaces
+- Global service and MCS-API support for non global(local) namespaces
+- Support Global/Local namespaces for non CRD modes of identity allocation 
+ 
+## Network Policy 
+### Cross-Cluster Communication and Network Policy Behavior
+
+Full support for **cross-cluster communication** and **network policy enforcement** is available **only when both the source (client) and destination (server) pods**—which may optionally be accessed via a service—**are part of a _global namespace_**. Network policies are not enforced if either the source or destination pod is in a **non-global namespace**.
+
+### Behavior When Pods Are Not in Global Namespaces
+Network policies in Cilium rely heavily on **identity labels** for policy matching. When the identity along with associated labels is not exported, the following behavior is observed:
+
+#### Tunnel Routing Mode
+
+- Identity ID is exported ✅  
+- Labels are NOT exported ❌  
+- **Result**: Destination cluster knows _who_ the source is (by ID) but not _what_ it is (labels). The policy applied through the labels is not enforced. 
```

Comment:
FYI, this is not correct, as it only applies to ingress policies. As for egress policies, the source cluster would not know the identity of the destination pod, neither the ID, nor the labels.